### PR TITLE
Add support for MIRACLE LINUX for hostname module

### PR DESCRIPTION
**Pull request by tSU-RooT**
##### SUMMARY
Add MIRACLE LINUX support for hostname module

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
lib/ansible/modules/hostname.py

##### ADDITIONAL INFORMATION
before:
```
PLAY [localhost] ***********************************************************************************************************************

TASK [Gathering Facts] *****************************************************************************************************************
ok: [localhost]

TASK [test-hostname-module] ************************************************************************************************************
fatal: [localhost]: FAILED! => {"changed": false, "msg": "hostname module cannot be used on platform Linux (Miraclelinux)"}

PLAY RECAP *****************************************************************************************************************************
localhost                  : ok=1    changed=0    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0
```
after:
```
PLAY [localhost] ***********************************************************************************************************************

TASK [Gathering Facts] *****************************************************************************************************************
ok: [localhost]

TASK [test-hostname-module] ************************************************************************************************************
ok: [localhost]

PLAY RECAP *****************************************************************************************************************************
localhost                  : ok=2    changed=0    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   

```

### CODE REVIEW

File: lib/ansible/modules/hostname.py [hunk 1]
```diff
@@ -779,6 +779,12 @@ class AmazonLinuxHostname(Hostname):
     strategy_class = RedHatStrategy
 
 
+class MiracleLinuxHostname(Hostname):
```
Comment (Akasurde) [inline lib/ansible/modules/hostname.py:782]: Please create a [changelog fragment](https://docs.ansible.com/ansible/latest/community/development_process.html#creating-a-changelog-fragment). See [this fragment](https://github.com/ansible/ansible/blob/c2f528b7686c9573aa8493357219e0bbe4416dd3/changelogs/fragments/reboot-change-default-boot-command.yaml) as an example.

**Review by Akasurde — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/hostname.py (hunk 1 above)
Comment (tSU-RooT) [reply lib/ansible/modules/hostname.py:782]: OK. I have created `75232-hostname-miraclelinux-support.yml`

**Review by Akasurde — APPROVED** [review]

**Comment by samdoran** [comment]
I thought we changed the module so that it no longer needed specific entries for every single distro. Looking at the code, I'm wondering if the systmed detection isn't working correctly if no subclass is found since `self.platform` will be `'Generic'`.

https://github.com/ansible/ansible/blob/a901ff2bc0621c1d1e16d8eccc182821dfec78f0/lib/ansible/modules/hostname.py#L153

I would rather work towards getting the module to be more dynamic and move away from needing to add an entry for every single distro.

**Review by Akasurde — CHANGES_REQUESTED** [review]
@samdoran Yes, I almost forgot that. @tSU-RooT Could you please update as per @samdoran's suggestion.

**Comment by tSU-RooT** [comment]
Hi, 
I have thought about suggestion.

If we will work for no-need to add code for every single distro,
I came up two ideas to fix this.
First idea is add configuration codes to `self.platform` for generic linux distros on Hostname class's `__init__`.

https://github.com/ansible/ansible/blob/a901ff2bc0621c1d1e16d8eccc182821dfec78f0/lib/ansible/modules/hostname.py#L145-L149

Second idea is remove `self.platform == 'Linux'` from elif,
since systemd managed system's platform is `Linux` usually and could use `hostnamectl`.

https://github.com/ansible/ansible/blob/a901ff2bc0621c1d1e16d8eccc182821dfec78f0/lib/ansible/modules/hostname.py#L153

Make sense?

**Pull request closed without merging**